packages/form-core/test-suites/NativeTextFieldMixin.suite.js

```diff
@@ -0,0 +1,63 @@
+import { LitElement } from '@lion/core';
+import { defineCE, expect, fixture, html, triggerFocusFor, unsafeStatic } from '@open-wc/testing';
+import { sendKeys } from '@web/test-runner-commands';
+import { getFormControlMembers } from '@lion/form-core/test-helpers';
+import { NativeTextFieldMixin } from '../src/NativeTextFieldMixin.js';
+
+/**
+ * @typedef {import('../types/FormControlMixinTypes').FormControlHost} FormControlHost
+ * @typedef {ArrayConstructor | ObjectConstructor | NumberConstructor | BooleanConstructor | StringConstructor | DateConstructor | 'iban' | 'email'} modelValueType
+ */
+
+/**
+ * @param {{tagString?: string, modelValueType?: modelValueType}} [customConfig]
+ */
+export function runNativeTextFieldMixinSuite(customConfig) {
+  const cfg = {
+    tagString: null,
+    ...customConfig,
+  };
+
+  describe('NativeTextFieldMixin', () => {
+    class NativeTextFieldClass extends NativeTextFieldMixin(LitElement) {
+      get slots() {
+        return {
+          // NativeTextFieldClass needs to have an _inputNode defined in order to work...
+          input: () => document.createElement('input'),
+        };
+      }
+    }
+
+    cfg.tagString = cfg.tagString ? cfg.tagString : defineCE(NativeTextFieldClass);
+    const tag = unsafeStatic(cfg.tagString);
+
+    it('preserves the caret position on value change', async () => {
+      const el = /** @type {NativeTextFieldClass} */ (await fixture(html`<${tag}></${tag}>`));
+      const { _inputNode } = getFormControlMembers(el);
+      await triggerFocusFor(el);
+      await el.updateComplete;
+      _inputNode.value = 'hello world';
+      _inputNode.selectionStart = 2;
+      _inputNode.selectionEnd = 2;
+      el.value = 'hey there universe';
+      expect(_inputNode.selectionStart).to.equal(2);
+      expect(_inputNode.selectionEnd).to.equal(2);
+    });
+
+    it.skip('move focus to a next focusable element after writing some text', async () => {
+      // TODO: this test did not fail before the fix for Safari bug: https://github.com/ing-bank/lion/issues/1415, so its not a good test.
+      const el = /** @type {NativeTextFieldClass} */ (await fixture(html`<${tag}></${tag}>`));
+      const { _inputNode } = getFormControlMembers(el);
+      await triggerFocusFor(el);
+      await el.updateComplete;
+      expect(document.activeElement).to.equal(_inputNode);
+      await sendKeys({
+        press: 'h',
+      });
+      await sendKeys({
```

Comment:
Maybe we can enable this test and instead try to check for a spy on `_setValueAndPreserveCaret` (that should not be called after the Tab)